swift-sdk/Internal/RequestCreator.swift

```diff
@@ -0,0 +1,317 @@
+//
+//
+//  Created by Tapash Majumder on 5/16/19.
+//  Copyright © 2019 Iterable. All rights reserved.
+//
+
+import Foundation
+
+enum IterableRequest {
+    case get(GetRequest)
+    case post(PostRequest)
+}
+
+struct GetRequest {
+    let path: String
+    let args: [String : String]?
+}
+
+struct PostRequest {
+    let path: String
+    let args: [String: String]?
+    let body: [AnyHashable : Any]?
+}
+
+// This is a stateless pure functional class
+// This will create IterableRequest
+// The Api Endpoint and request endpoint is not defined yet
+struct RequestCreator {
+    let apiKey: String
+    let auth: Auth
+    
+    func createUpdateEmailRequest(newEmail: String) -> Result<IterableRequest, IterableError> {
+        var body: [String : Any] = [
+            AnyHashable.ITBL_KEY_NEW_EMAIL: newEmail
+        ]
+        
+        if let email = auth.email {
+            body[AnyHashable.ITBL_KEY_CURRENT_EMAIL] = email
+        } else if let userId = auth.userId {
+            body[AnyHashable.ITBL_KEY_CURRENT_USER_ID] = userId
+        } else {
+            ITBError("Both email and userId are nil")
+            return .failure(IterableError.general(description: "Both email and userId are nil"))
+        }
+        
+        return .success(.post(createPostRequest(path: .ITBL_PATH_UPDATE_EMAIL, body: body)))
+    }
+    
+    func createReqisterTokenRequest(hexToken: String,
```

Comment:
```suggestion
    func createRegisterTokenRequest(hexToken: String,
```

Accidental misspelling?